sync/sync.go

```diff
@@ -686,15 +686,15 @@ func MakeBranchPRs(f *Flags, dir string, entry *ConfigEntry) ([]SyncResult, erro
 			continue
 		}
 
-		auther, err := f.Auth.NewAuther()
+		auther, err := f.Auth.NewAuther(*f.DryRun)
 		if err != nil {
 			if !errors.Is(err, githubutil.ErrNoAuthProvided) {
 				return nil, err
 			}
 			c.SkipReason = fmt.Sprintf("provided auth doesn't support PR submission: %v", err)
 			continue
 		}
```

Comment:
The bool changes behavior pretty significantly--I think this block is the clearest at showing why a `githubutil.ErrNoAuthProvided` must happen even in a dry run. The command itself needs to determine what to do about a dry run: sometimes a lot of code needs to be skipped that can't be done anonymously at all (whether due to rate limits or permissions).

This is more of what I was thinking: https://github.com/dagood/go-infra/commit/898934329d5b2981225905071e22a58d489c50db

Reply:
agreed, I've applied your commit, let me know what you think